tsconfig.base.json

```diff
@@ -0,0 +1,15 @@
+{
+  "compilerOptions": {
+    "module": "node16",
```

Comment:
that's a bit confusing but correct: [`module` defines the module system](https://www.typescriptlang.org/tsconfig#module) (which supports ESM since node v16 so called "node16", versus only supporting CommonJS called "node"), not the target runner (node v20 in our case, or browsers).